internal/descriptor/provider/provider.go

```diff
@@ -44,7 +45,8 @@ func (c *CachedDescriptorProvider) GetDescriptor(template *v1beta2.ModuleTemplat
 
 		return desc, nil
 	}
-	key := cache.GenerateDescriptorKey(template)
+	key := c.GenerateDescriptorKey(template.Name, template.GetVersion())
```

Comment:
To Reviewer:

The parameter should be refactored in this issue https://github.com/kyma-project/lifecycle-manager/issues/2601, not from template, instead with ocmComponentName and ocm version